### ISSUE TITLE
Using --snake-case-field produces names with double underscores

### ISSUE THREAD

**Issue by wardy3** [koxudaxi/datamodel-code-generator#1014]
Using the `--snake-case-field` switch works well for "standard" cases but can converts name with an `_` into double `__`.

Eg, 

```json
{
    "snake_case": "Snake case",
    "camelCase": "Camel case",
    "kebab-case": "Kebab case",
    "PascalCase": "Pascal case",
    "UPPER_CASE": "Upper case",
    "Dev_Info": "example 1",
    "CLONE_Device": "example 2"
}
```

and the command

```bash
datamodel-codegen --target-python-version 3.9 --enable-faux-immutability --reuse-model --use-subclass-enum  --input-file-type json --input test.json  --snake-case-field                            
```

creates 
```python
class Model(BaseModel):
    class Config:
        allow_mutation = False

    snake_case: str
    camel_case: str = Field(..., alias='camelCase')
    kebab_case: str = Field(..., alias='kebab-case')
    pascal_case: str = Field(..., alias='PascalCase')
    upper_case: str = Field(..., alias='UPPER_CASE')
    dev__info: str = Field(..., alias='Dev_Info')
    clone__device: str = Field(..., alias='CLONE_Device')
```

Note the last two have doubled up on underscores.

I would expect the last two to be called `dev_info` and `clone_device`

**Version:**
 - OS: Mac OS 13.1
 - Python version: 3.9.0
 - datamodel-code-generator version: 0.15.0

thanks for the great library!

**Comment by koxudaxi** [comment]
@wardy3 
I have released the new version as `0.16.1`
Thank you very much!!

**Comment by tobiasfeil** [comment]
You're an ABSOLUTE LEGEND :star_struck:
